1) In a best of 9 frame snooker match between Player A and Player B, it is estimated that the probability for Player A to win the first frame is 55%. Using this information:
a) estimate the probability for Player A to win the match,

the probability of A winning first game is 55%
We can assume for the remaining game the probability of A winning is Pa. We can assume it to be 50%
To win the match A needs to win atleast 5 games out of 9.
Therefore the probability of A winning = 0.55 + (Pa)**4

In [5]:
Pa = 0.55 + 0.5**4 
print(Pa)

0.6125


Hence, the probability of A winning an individual game is 61.25%

b) estimate the probability for there to be at least 8 frames played in the match

For atleast 8 games A or B needs to win atleast 4 games of which A winning first game has probability of 55%.

some scenario. (AB) refers to any of them winning the game to win 5 games and win the match.
AAAABBB(AB)
AAABBBB(AB)
AAAABBBB(AB)




In [28]:
eig = 0.55 + 0.5**3 * 0.5**4
print("the probability for there to be at least 8 frames played in the match " + str(round(round(eig,2)*100))+"%")


the probability for there to be at least 8 frames played in the match 56%


c) what would be an appropriate level of odds (in decimal format) to offer for Player B to win the match.


In [17]:
Pb = 1 - Pa

WinStake = 1/Pb

print("an appropriate level of odds (in decimal format) to offer for Player B to win the match is " + str(round(WinStake,2)))

an appropriate level of odds (in decimal format) to offer for Player B to win the match is 2.58


2) In a football match between Team A and Team B, Team A is estimated to have a goals expectancy of 1.5 goals and Team B is estimated to have a goals expectancy of 1.25 goals. Using the Poisson Distribution:
a) estimate the probability for Team A to win the match,

In [33]:
from scipy.stats import poisson
teamA = 1.5
teamB = 1.25

teamAWins = 1- poisson.cdf(k = teamB, mu = teamA)
print("the probability for Team A to win the match is "+ "{:.2f}".format(teamAWins*100)+"%")

the probability for Team A to win the match is 44.22%


estimate the probability for there to be at least 2.5 goals in the match

 In a tennis match, the probability of Player A winning a given point on serve in the first game of the match is 0.52. Find the probability that Player B wins the first game, given player A is serving.

In [34]:
Pa = 0.52

Pb = 1 - Pa

print(Pb)

0.48
